src/LiveComponent/doc/index.rst

```diff
@@ -2255,26 +2259,88 @@ real component once the page has loaded.
+
+To add some loading content before the real component is loaded, you can use the
+``loading-template`` option to point to a template:
+
+.. code-block:: html+twig
+
+    {# With the HTML syntax #}
+    <twig:SomeHeavyComponent defer loading-template="spinning-wheel.html.twig" />
+
+    {# With the component function #}
     {{ component('SomeHeavyComponent', { defer: true, loading-template: 'spinning-wheel.html.twig' }) }}
 
 Or override the ``loadingContent`` block:
 
-.. code-block:: twig
+.. code-block:: html+twig
 
+    {# With the HTML syntax #}
+    <twig:SomeHeavyComponent defer>
+        <twig:block name="loadingContent">Custom Loading Content...</twig:block>
+    </twig:SomeHeavyComponent>
+
+    {# With the component tag #}
     {% component SomeHeavyComponent with { defer: true } %}
         {% block loadingContent %}Loading...{% endblock %}
     {% endcomponent %}
 
-To change the initial tag from a ``div`` to something else, use the ``loading-tag`` option:
+Both ``loading-template`` and ``loadingContent`` can be used in the template
+_calling_ the component. But if want to set this content only once, you can add a
+``placeholder`` macro directly in the component template.
 
-.. code-block:: twig
+.. versionadded:: 2.17.0
 
-    {{ component('SomeHeavyComponent', { defer: true, loading-tag: 'span' }) }}
+    Defining a placeholder macro into the component template was added in Live Components 2.17.0.
+
+In the component template, define a ``placeholder`` macro, outside of the
+component's main content. This macro will be called when the component is deferred:
+
+.. code-block:: html+twig
+
+    {# templates/recommended-products.html.twig #}
+    <div {{ attributes }}>
+        {# This will be rendered when the component is fully loaded #}
+        {% for product in this.products %}
+            <div>{{ product.name }}</div>
+        {% endfor %}
+    </div>
+
+    {% macro placeholder(props) %}
+        {# This content will (only) be rendered as loading content #}
+        <span class="loading-row"></span>
+    {% endmacro %}
+
+The ``props`` argument contains the props passed to the component.
+You can use it to customize the placeholder content. Let's say your
+component shows a certain number of products (defined with the ``size``
+prop). You can use it to define a placeholder that shows the same
+number of rows:
+
+.. code-block:: html+twig
+
+    {# In the calling template #}
+    <twig:RecommendedProducts size="3" defer />
+
+.. code-block:: html+twig
+
+    {# In the component template #}
+    {% macro placeholder(props) %}
+        {% for i in 1..props.size %}
+            <div class="loading-product">
+                ...
+            </div>
+        {% endfor %}
+    {% endmacro %}
+
+If you style the placeholder content (giving the rows a specific height),
+you can then avoid layout shifts when the component is fully loaded (and
+the loading content replaced by the real content).
```

Comment:
I think it’s a bit too long - the first example is probably enough. 

Also, I’m wondering if this should license the FIRST option of customizing loading content. I’d imagine that I would more often embed the loading content inside the template vs in the calling template.

so logic would be:

- to customize loading, you can add this macro
- Else, if you want to control loading when calling the component, use the block trick.
- And, btw, you can control the loading tag for edge cases, like when your element is a tr

Reply:
I've reordered all .. tell me if better